tests/lean/run/eq23.lean

```diff
@@ -10,7 +10,7 @@ with tree_list :=
 namespace tree_list
 
 definition len {A : Type} : tree_list A → nat
-| len (nil A)    := 0
```

Comment:
I believe this _should_ have been illegal even under the old rules, but `validate_pattern` didn't catch it.